### PR TITLE
Native Qwen3.5 support

### PR DESCRIPTION
**Pull request by tomiock**
Creation of a native pytorch Qwen3.5 model, dropping the `transformers` model definition.

### CODE REVIEW

**Comment by tomiock** [comment]
Model running, wrong loss value, but is converging to lower value.

Compile diff times:
- 0.162s with
- 0.175 without

image

**Comment by tomiock** [comment]
Loss parity issue vs HF implementation:
```
  Root cause of the training-loss mismatch: HF's Qwen3_5Model.forward (transformers 5.5.3) does

  linear_attn_mask = self._update_linear_attn_mask(attention_mask, past_key_values)

  and feeds linear_attn_mask (not the varlen cu_seqlens) to every GatedDeltaNet. HF's linear-attention path therefore is not varlen-aware: when two
  samples are packed into one row, sample A's recurrent state leaks into sample B. Our native model is correct — it threads cu_seqlens into FLA's
  chunk_gated_delta_rule and runs the causal Conv1d per-segment — so the native and HF losses are expected to diverge on packed batches. Native (lower
   loss) is the more correct value.

  Verification:
  - test_qwen3_5_text_parity: bf16 logits match HF (max 0.16).
  - test_qwen3_5_varlen_pack: packed sample B's argmax == solo sample B's argmax.
  - test_qwen3_5_loss_parity:
    - Check 1 (single-sample, no packing): ours vs HF max |diff| = 5e-3.
    - Check 2 (packed + padded, native vs native solo): argmax matches for both samples (internal consistency).

  Takeaway: on packed training, "native < HF" is the correct direction — HF is overestimating loss because of cross-sample state leakage. You won't
  get numeric parity with HF on packed batches unless you also patch HF's _update_linear_attn_mask to be varlen-aware. For a fair side-by-side, either
   run both with batch_size=1 (no packing) or compare native vs a patched HF.
```

if a run with a single sample performs okay perhaps we have nothing to worry

**Comment by tomiock** [comment]
image

Loss matches perfectly with a single sample.
On large runs the native loss is lower than the HF loss, will create issue and pospone.
